src/Lexer.php

```diff
@@ -337,14 +337,15 @@ private function lexExpression(): void
             $this->pushToken(Token::SPREAD_TYPE, '...');
             $this->moveCursor('...');
         }
-        // arrow function
-        elseif ('=' === $this->code[$this->cursor] && ($this->cursor + 1 < $this->end) && '>' === $this->code[$this->cursor + 1]) {
-            $this->pushToken(Token::ARROW_TYPE, '=>');
-            $this->moveCursor('=>');
-        }
         // operators
         elseif (preg_match($this->regexes['operator'], $this->code, $match, 0, $this->cursor)) {
-            $this->pushToken(Token::OPERATOR_TYPE, preg_replace('/\s+/', ' ', $match[0]));
+            $operator = preg_replace('/\s+/', ' ', $match[0]);
+            $type = Token::OPERATOR_TYPE;
+            // to be removed in 4.0
```

Comment:
This removal will likely make things weird for the token stream (which will impact custom token parsers). After this removal, `(` would generator an `OPERATOR_TYPE` token, while `)` will not (will it stay a `PUNCTUATION_TYPE` ? it depends how the new `PUNCTUATION` constant will look like)

Reply:
Code removed as not needed anymore.